### PR TITLE
Updated Diagnostics.Log

### PR DESCRIPTION
**Pull request by wrestledBearOnce**
Shrunk to 3 log level (Debug, Warning, Error).
Fixed crash when the object which should be logged was null.
Moved project specific log level to Dir.Build.props

### CODE REVIEW

**Comment by wrestledBearOnce** [comment]
The constants per project files are being overwritten by the build.props instead of appended.
One has to tinker around with all project files to define a $(constants) to be able to append stuff like this: $(constants);PLATFORM_DESKTOP;TRACE ...

File: src/Base/Core/Diagnostics.cs [hunk 1]
```diff
@@ -138,6 +123,10 @@ static Diagnostics()
         {
             // Overwrite file
             File.Create(_fileName).Close();
```
Comment (ASPePeX) [inline src/Base/Core/Diagnostics.cs]: This doesn't work for the Android build but I have no out of the box solution atm.

**Comment by ASPePeX** [comment]
I'm no longer sure if we need all the compiler constants. In the end it comes down to is it a DEBUG build or not. So I think we could just as well use `if (!)DEBUG`.

File: src/Base/Core/Diagnostics.cs [hunk 2]
```diff
@@ -123,6 +123,10 @@ static Diagnostics()
...
+
+#if LOG_DEBUG
+            SetMinConsoleLoggingSeverityLevel(SeverityLevel.DEBUG);
```
Comment (wrestledBearOnce) [inline src/Base/Core/Diagnostics.cs]: Just set _minLogLevelConsole to SeverityLevel.DEBUG

**Pull request merged**